src/modules/auth/auth.service.ts

```diff
@@ -1,87 +1,85 @@
-import { JwtService } from '@nestjs/jwt';
-import { HttpStatus, Injectable } from '@nestjs/common';
+import {
+  HttpStatus,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
 import * as bcrypt from 'bcrypt';
 
 import { UserService } from '../user/user.service';
-import { RefreshTokenService } from './refresh-token.service';
 
-import { TokenPair } from './interfaces/token-pair.interface';
-import { AccessToken } from './interfaces/access-token.interface';
-import { AuthenticatedUserInfo } from './interfaces/authenticated-user-info.interface';
+import { TokenService } from './token.service';
+
+import { User } from 'src/database/entities';
+import { CreateUserDto } from '../user/dto/create-user.dto';
+
+import { TokenPair } from './auth.interface';
+import { UserPayload } from './auth.interface';
 
 @Injectable()
 export class AuthService {
   constructor(
-    private jwtService: JwtService,
     private userService: UserService,
-    private refreshTokenService: RefreshTokenService,
+    private tokenService: TokenService,
   ) {}
 
-  async issueTokenPair(user: AuthenticatedUserInfo): Promise<TokenPair> {
-    const payload = { username: user.username, id: user.id };
-
-    const token = this.jwtService.sign(payload);
-    const refreshToken = await this.refreshTokenService.generate(payload.id);
-
-    return {
-      token,
-      refreshToken,
-      tokenType: 'Bearer',
-    };
+  generatePayload({ id, username, email }): UserPayload {
+    return { id, username, email };
   }
 
-  async validateUser(
-    email: string,
-    pass: string,
-  ): Promise<AuthenticatedUserInfo | null> {
-    const user = await this.userService.findByEmail(email);
-
-    if (user && bcrypt.compare(pass, user.password)) {
-      const { id, username, email } = user;
-      const authenticatedUserInfo = { id, username, email };
+  async issueTokenPair(user: UserPayload | User): Promise<TokenPair> {
+    const tokens = this.tokenService.generateTokens(this.generatePayload(user));
```

Comment:
Is it need to create function which returns its params, i mean generatePayload return user which you are passing in as param.